frontend/src/auth/ProtectedRoute.tsx

```diff
@@ -0,0 +1,9 @@
+import { useAuth0 } from '@auth0/auth0-react';
+import { Navigate, Outlet } from 'react-router-dom';
+
+const ProtectedRoute = () => {
+  const { isAuthenticated } = useAuth0();
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/24#issuecomment-2381262689>#f63685</a><span> - 09/29/2024, 08:36 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The component doesn't handle loading states, which may lead to a flash of incorrect content
</div>
<div id="fix">
<b>Fix:</b> Add a loading state check using the isLoading property from useAuth0. Render a loading indicator while authentication status is being determined.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,9 +1,13 @@
  import { useAuth0 } from '@auth0/auth0-react';
  import { Navigate, Outlet } from 'react-router-dom';
 
  const ProtectedRoute = () => {
 -  const { isAuthenticated } = useAuth0();
 +  const { isAuthenticated, isLoading } = useAuth0();
 +
 +  if (isLoading) {
 +    return <div>Loading...</div>;
 +  }
 +
    return isAuthenticated ? <Outlet /> : <Navigate to="/" replace />;
  };
 
  export default ProtectedRoute;
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged